aptos-move/block-executor/src/executor.rs

```diff
@@ -572,45 +565,41 @@ where
                                 finalized_groups.push((group_key, metadata_op, finalized_group));
                             },
                             Err(err) => {
-                                maybe_err = Some(err);
+                                maybe_code_err = Some(err.into());
                                 break;
                             },
                         }
-                        if maybe_err.is_some() {
+                        if maybe_code_err.is_some() {
                             break;
                         }
                     }
                 }
             }
 
-            last_input_output.record_finalized_group(txn_idx, finalized_groups);
+            // We return an error (leads to halting the execution) in the following cases:
+            // 1) Code invariant violation.
+            // 2) We detect module read/write intersection
+            // 3) A transaction triggered an Abort
 
-            maybe_err = maybe_err.or_else(|| last_input_output.maybe_execution_error(txn_idx));
+            last_input_output.record_finalized_group(txn_idx, finalized_groups);
 
-            // We `halt` the execution in the following 4 cases:
-            // 1) We detect module read/write intersection
-            // 2) A transaction triggered an Abort
-            // 3) All transactions are scheduled for committing
-            // 4) We skip_rest after a transaction
+            if let Some(err) = maybe_code_err.or_else(|| last_input_output.module_rw_intersection())
```

Comment:
What about resource group error?

Reply:
the bcs one is a dedicated error type now, so we can do specialized handling in the fallback, everything else I wanted to handle as other code errors, that's why I re-used the code_invariant_error - and they should follow the same flow here? (but taking care of the wrong/specialized message for delayed fields)

Reply:
I see, thanks!